packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SVGComponents/VisXZoom/VisXZoom.js

```diff
@@ -31,16 +64,69 @@ function VisXZoom({
   zoomConfiguration = defaultZoomConfig,
   zoomingComponent,
   zooming = false,
-  ...props
 }) {
   const { onKeyZoom } = useKeyZoom()
   const zoomRef = useRef(null)
+  const wheelEventLayer = useRef(null)
 
   useEffect(function setCallbacks() {
     setOnPan(handleToolbarPan)
     setOnZoom(handleToolbarZoom)
   }, [setOnPan, setOnZoom])
 
+  function wheelDelta(event) {
+    const zoomValue = (-event.deltaY > 0) ? 1.1 : 0.9
+    return { scaleX: zoomValue, scaleY: zoomValue }
+  }
+
+  const wheelHandler = zooming ? (e) => zoomRef.current?.handleWheel(e) : DEFAULT_HANDLER
+  const throttledWheelHandler = throttle(wheelHandler, zoomConfiguration?.onWheelThrottleWait)
+  
+  function onWheel(event) {
+    /* Default behaviour for subject viewers that don't scroll vertically.
+    Cancel the default event (ignored unless the listener explicitly
+    sets passive: false) and call the wheel handler with a throttled delay.
+    */
+    if (!allowsScrolling) {
+      /* event.preventDefault() will throw a warning in the browser
+      for passive events. To avoid that, use addEventListener with 
+      { passive: false }.
+      */
+      event.preventDefault()
+      return throttledWheelHandler(event)
+    }
+    
+    /* Zoom if the zoom hot key is pressed.
+    Passive event listeners, like React's onWheel or default browser wheel events,
+    won't prevent the default wheel scroll, so style the document body to stop
+    scrolling and remove scrollbars.
+    */
+    if (event[ZOOM_HOT_KEY]) {
+      /* Disable scroll in Safari (or for passive wheel events.) */
+      document.body.style.overflow = 'hidden'
+      const scrollbarWidth = window.innerWidth - document.documentElement.clientWidth
+      if (scrollbarWidth) {
+        document.body.style.paddingRight = `${scrollbarWidth}px`
+      }
```

Comment:
If the scrollbar is hidden before the scrollbar width is calculated, then the calculated `scrollbarWidth` will always be zero, and there's a horizontal layout shift when a user begins scrolling with the hot key.

Can the scrollbar width be calculated first, then `overflow: hidden` ? Doesn't have to be exactly like the following, but something like:

```suggestion
      /* Calculate width of scrollbar */
      const scrollbarWidth = window.innerWidth - document.documentElement.clientWidth
      
      /* Disable scroll in Safari (or for passive wheel events), hide scrollbar */
      document.body.style.overflow = 'hidden'
      
      /* Add right padding for the width of the scrollbar to prevent horizontal layout shift */
      if (scrollbarWidth) {
        document.body.style.paddingRight = `${scrollbarWidth}px`
      }
```

Reply:
Sure, that should be fine (but do check in desktop Safari.)

The underlying problem here is that `ZoomEventLayer` uses React's `onWheel`, which fires passive events. 

If you update `ZoomEventLayer` to fire active events, with `passive: false`, that might remove some of these hacks that work around passive wheel events. That would be a PR in its own right, however.

Reply:
Fixed!